### ISSUE TITLE
Feature(BulletForge): Training Data Pipelines Infrastructure

### ISSUE THREAD

**Issue by justinthelaw** [justinthelaw/smarter-bullets#52]
**Is your feature request related to a problem? Please describe.**
We need a structured way to prepare raw data and send it to OpenAI's API for fine-tuning.

**Describe the solution you'd like**
Generate 1 or more Jupyter Notebooks that can be run to take-in raw training data and produce a fine-tuned version of a GPT3 base model.

**Describe alternatives you've considered**
N/A

**Additional context**
In the `models/` directory, there are resources to get started. The OpenAI API documentation is also an important area to do more research in if the developer is unsure of the syntax or request-response API schemas.

**Comment by justinthelaw** [comment]
Added a basic scraping component (Python) to the branch for data gathering inside of the aforementioned model directory. Scraper `crawl()` is going to URLs beyond the base URL. Need to scope-limit script and improve error handling and user experience.

**Comment by justinthelaw** [comment]
The initial setup of the fine-tuning Jupyter Notebook is complete, and a somewhat robust bullet scraper has been built with a complexity level of 1 (simple HTML). Remaining tasks:

- The scraper needs higher complexity and testing for other bullet repositories. 
- Fine-tuning data (JSONL) needs to be created, mainly the prompt portions as compared to the completions sections

**Comment by justinthelaw** [comment]
The `scraper/` directory has been reorganized and integrated in the `models/` directory, and implemented within the `training-data.ipynb`. Next step is to build out the training data pipeline using ChatGPT to generate the prompt for the scraped completions.

**Comment by justinthelaw** [comment]
Scraper and Consolidator are done now. The 44K+ line training data JSONL file needs prompts (or labels) before it is then vectorized and encoded for training on a model.

**Comment by justinthelaw** [comment]
Switched entire repo over to T5 base models and starting work on fine-tuning. Pulled T5 base model. Fixed entire `forge/` directory to match best practices for Machine Learning.

**Comment by justinthelaw** [comment]
Data strategy is complete and documented. Moving on to completing data set creation and model training features.
